### PR TITLE
Focus on `ModalCurtain` root instead of first focusable element (#156)

### PR DESCRIPTION
**Pull request by jichan-thumbtack**
- Upgraded Enzyme to easily test for `:focus` state (https://github.com/airbnb/enzyme/pull/1965)

- Upgraded forcus-trap-react (https://github.com/davidtheclark/focus-trap-react/blob/master/CHANGELOG.md) One breaking change seems to be that <FocusTrap> no longer wraps children in an extra div; instead it expects a single top-level child (better that it's one less nested `<div>`?)

### CODE REVIEW

File: packages/thumbprint-react/components/ModalCurtain/index.jsx [hunk 1]
```diff
@@ -82,39 +84,53 @@ export default class ModalCurtain extends React.Component {
...
-                {shouldRenderEscListener && <EscListener onEscPress={onCloseClick} />}
+                {/**
+                    Use tabIndex="-1" to allow programmatic focus (as initialFocus node for <FocusTrap>)
+                    but not be tabbable by user.
+                */}
+                <div
+                    role="dialog"
+                    aria-label={accessibilityLabel}
+                    tabIndex="-1"
+                    ref={div => {
```
Comment (danoc) [inline packages/thumbprint-react/components/ModalCurtain/index.jsx]: Would be great to use React's latest API for creating ref's:
https://reactjs.org/docs/refs-and-the-dom.html#creating-refs

Comment (jichan-thumbtack) [reply packages/thumbprint-react/components/ModalCurtain/index.jsx]: Focus-trap-react only supports callback ref for now :( https://github.com/davidtheclark/focus-trap-react/blob/master/src/focus-trap-react.js#L82) Support for object ref is under CR (https://github.com/davidtheclark/focus-trap-react/pull/41) so we can switch after that goes through?

Another thing that I noticed when testing modals on screenreader was that we need aria-labelledby and aria-describedby for screenreader to announce contents in `ModalHeader` and `ModalDescription`. (https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html) Is this something already being worked on?

Comment (danoc) [reply packages/thumbprint-react/components/ModalCurtain/index.jsx]: Thanks for digging into that.

After learning more about VoiceOver, I think this _is_ working. Try this:

1. Turn on VoiceOver and visit https://deploy-preview-163--thumbprint.netlify.com/components/modal-default/react/
2. Open a modal.
3. Hold Control + Option + Shift and then press the down arrow. This should take you to the title.
4. Hold Control + Option + Shift and then use the left and right arrow keys. This should cycle through the DOM nodes in the modal.

We could use `ModalHeader` and `ModalDescription` but there are plenty of times that people don't use those components and roll-their own styles.

Comment (danoc) [reply packages/thumbprint-react/components/ModalCurtain/index.jsx]: One thing I noticed in my testing is that the modals are accessible to screen readers even when they're closed. I wonder if we should add `aria-hidden="true"` if it's closed.

That can be a separate task, but I'd be curious to know what you think. (I looked into `aria-modal` but was still able to navigate outside of the modal in VoiceOver 🤷‍♂️.)

**Pull request merged**